diff --git a/src/Web/Theory.hs b/src/Web/Theory.hs
index 7abb21b745363013c4cabc5ba9391d0cd293d322..6441e44fea5a0b5f4b09bf6684fac113347113b8 100644
--- a/src/Web/Theory.hs
+++ b/src/Web/Theory.hs
@@ -818,7 +818,9 @@ subDiffProofSnippet renderUrl tidx ti lemma proofPath ctxt prf =
     refSubCase (name, prf') =
         [ withTag "h4" [] (text "Case" <-> text name)
         , maybe (text "no proof state available")
-                (const $ refDotInteractiveDiffPath renderUrl tidx (DiffTheoryDiffProof lemma (proofPath ++ [name])) False)
+                (\se -> if nonEmptyGraphDiff se
+                   then refDotInteractiveDiffPath renderUrl tidx (DiffTheoryDiffProof lemma (proofPath ++ [name])) False
+                   else emptyDoc)
                 (dpsInfo $ root prf')
         ]
 
